R/transition_risk_profile.R

```diff
@@ -0,0 +1,128 @@
+#'   `profile_emissions()`.
+#' @param sector_profile Nested data frame. The output of `profile_sector()`.
+#' @param pivot_wider Logical. Pivot the output at company level to a wide
+#'   format?
+#' @param emissions_profile_products A dataframe
+#' @param all_uuids_scenario_sectors A dataframe
+#' @param scenarios A dataframe
+#'
+#' @return A data frame with the column `companies_id`, and the nested
+#'   columns`product` and `company` holding the outputs at product and company
+#'   level.
+#' @export
+#'
+#' @examples
+#' library(readr, warn.conflicts = FALSE)
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(tiltToyData, warn.conflicts = FALSE)
+#' library(tiltTransitionRisk, warn.conflicts = FALSE)
+#'
+#' set.seed(123)
+#' restore <- options(list(
+#'   readr.show_col_types = FALSE,
+#'   tiltIndicatorAfter.output_co2_footprint = TRUE
+#' ))
+#'
+#' toy_emissions_profile_products_ecoinvent <- read_csv(toy_emissions_profile_products_ecoinvent())
+#' toy_emissions_profile_any_companies <- read_csv(toy_emissions_profile_any_companies())
+#' toy_sector_profile_any_scenarios <- read_csv(toy_sector_profile_any_scenarios())
+#' toy_sector_profile_companies <- read_csv(toy_sector_profile_companies())
+#' toy_europages_companies <- read_csv(toy_europages_companies())
+#' toy_ecoinvent_activities <- read_csv(toy_ecoinvent_activities())
+#' toy_ecoinvent_europages <- read_csv(toy_ecoinvent_europages())
+#' toy_ecoinvent_inputs <- read_csv(toy_ecoinvent_inputs())
+#' toy_isic_name <- read_csv(toy_isic_name())
+#'
+#' emissions_profile <- profile_emissions(
+#'   companies = toy_emissions_profile_any_companies,
+#'   co2 = toy_emissions_profile_products_ecoinvent,
+#'   europages_companies = toy_europages_companies,
+#'   ecoinvent_activities = toy_ecoinvent_activities,
+#'   ecoinvent_europages = toy_ecoinvent_europages,
+#'   isic = toy_isic_name
+#' )
+#'
+#' sector_profile <- profile_sector(
+#'   companies = toy_sector_profile_companies,
+#'   scenarios = toy_sector_profile_any_scenarios,
+#'   europages_companies = toy_europages_companies,
+#'   ecoinvent_activities = toy_ecoinvent_activities,
+#'   ecoinvent_europages = toy_ecoinvent_europages,
+#'   isic = toy_isic_name
+#' )
+#'
+#' emissions_profile_products <- read_csv(toy_emissions_profile_products_ecoinvent())
+#' all_uuids_scenario_sectors <- read_csv(toy_all_uuids_scenario_sectors())
+#' scenarios <- read_csv(toy_sector_profile_any_scenarios())
+#'
+#' output <- transition_risk_profile(emissions_profile,
+#'   sector_profile,
+#'   pivot_wider = FALSE,
+#'   emissions_profile_products,
+#'   all_uuids_scenario_sectors,
+#'   scenarios
+#' )
+#'
+#' output |> unnest_product()
+#'
+#' output |> unnest_company()
+transition_risk_profile <- function(emissions_profile,
+                                    sector_profile,
+                                    emissions_profile_products,
+                                    all_uuids_scenario_sectors,
```

Comment:
These two names seem a bit awckward. For inspiration see the argument names in the existing functions ([example](https://2degreesinvesting.github.io/tiltIndicatorAfter/reference/profile_emissions_upstream.html#ref-usage)). Typically each name captures one single most important aspect of the data, or two such aspects when the dataset maps one dataset to another dataset. 

e.g. 

* `ecoinvent_activities` may have lots of columns but most importantly it has ecoinvent activities.
* `ecoinvent_europages` maps data from europages to data from europages.

Reply:
@maurolepore The `all_uuids_scenario_sectors` has all the ecoinvent activities mapped to scenario sectors. Should I name it as `ecoinvent_all_scenario_sectors`? What would you suggest? 

I will rename `emissions_profile_products` to `co2` because they are the same.

Reply:
How about `activities_sectors`? 

"Ecoinvent" seems implicit in "activities" and "scenario" seems implicit in "sectors".

Reply:
Also I realize a name including "ecoinvent" is somewhat brittle. What we really care about is "activities" -- the fact that we get them from "ecoinvent" is a detail that may change if we find a better provider some day.

Reply:
Sector can be of any type like isic, tilt, and scenarios. Hence, we need to specify scenarios. How about `activities_scenario_sec`?

Reply:
Also, we have another dataset which have some activities with scenario sectors. The term `all` is used to specify that we have now all activities which are mapped with scenario sectors.

Reply:
Thanks for your opinion! The new dataset name should show for what purpose it is created. It is created to map all the ecoinvent activities to scenario sectors. Hence, I will rename it `all_activities_scenario_sectors`. I hope we are good with it :)